File: ProjetComparatif/Csharp/Tache3/Tache3_AnalyseTexte.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

class Tache3_AnalyseTexte
{
    static void Main()
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        string filePath = "texte.txt";

        if (!File.Exists(filePath))
        {
            Console.WriteLine($"Fichier introuvable : {filePath}");
            return;
        }

        string texte = File.ReadAllText(filePath).ToLower();

        // Extraire les mots avec une expression régulière
        var mots = Regex.Matches(texte, @"\b\w+\b")
                        .Cast<Match>()
                        .Select(m => m.Value)
                        .ToList();

        int nbTotal = mots.Count;

        // Compter les fréquences
        var frequence = mots
            .GroupBy(m => m)
            .ToDictionary(g => g.Key, g => g.Count());

        // Extraire les 10 mots les plus fréquents
        var top10 = frequence
            .OrderByDescending(kvp => kvp.Value)
            .Take(10);

        // Affichage
        Console.WriteLine($"Nombre total de mots : {nbTotal}\n");
        Console.WriteLine("10 mots les plus fréquents :\n");
        foreach (var kvp in top10)
        {
            Console.WriteLine($"{kvp.Key} : {kvp.Value}");
        }
        // Arrêt du chronomètre et affichage du temps d'exécution
        stopwatch.Stop();
        Console.WriteLine($"\n⏱ Temps d'exécution : {stopwatch.Elapsed.TotalMilliseconds:F3} millisecondes");
    }
}

File: ProjetComparatif/python/tache3_analyse_texte.py
import time
from collections import Counter
import re

start = time.time()

# Lire le fichier texte
with open("texte.txt", "r", encoding="utf-8") as f:
    texte = f.read().lower()  # Minuscule pour éviter les doublons "Le" vs "le"

# Nettoyer le texte (enlever ponctuation, conserver les mots)
mots = re.findall(r'\b\w+\b', texte)

# Compter les mots
nb_total = len(mots)
frequence = Counter(mots)
top10 = frequence.most_common(10)

# Affichage des résultats
print(f"Nombre total de mots : {nb_total}\n")
print("10 mots les plus fréquents :\n")
for mot, count in top10:
    print(f"{mot} : {count}")

end = time.time()
print(f"\n⏱ Temps d'exécution : {(end - start) * 1000:.3f} millisecondes")


